### PR TITLE
Make -1 the invalid view

### PR DESCRIPTION
**Pull request by ashamis**

### CODE REVIEW

**Comment by eddyashton** [comment]
This definition is just trying to describe what the existing consensus does. In `raft.h` `get_term_internal`, we return 0 for an unknown index.

**Comment by ashamis** [comment]
> This definition is just trying to describe what the existing consensus does. In `raft.h` `get_term_internal`, we return 0 for an unknown index.

I updated the PR to make this dependancy explicit.

File: src/node/rpc/tx_status.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ namespace ccf
      {TxStatus::Committed, tx_status_to_str(TxStatus::Committed)},
      {TxStatus::Invalid, tx_status_to_str(TxStatus::Invalid)}});
 
-  constexpr size_t VIEW_UNKNOWN = 0;
+  constexpr size_t VIEW_UNKNOWN = -1;
```
Comment (achamayou) [inline src/node/rpc/tx_status.h:51]: Let's have the View be signed everywhere including in the tx rpc then.

**Comment by eddyashton** [comment]
I'm all for using the named value in more places, but I don't see why we're changing the invalid value from 0 to -1?

**Review by achamayou — APPROVED** [review]

**Comment by ashamis** [comment]
> I'm all for using the named value in more places, but I don't see why we're changing the invalid value from 0 to -1?

We are freeing up 0 to allow that to be a valid view. This is useful when we determine the primary using the view.

**Pull request merged**